lambdas/gro-pull-file/src/main/java/uk/gov/di/data/lep/PullFile.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.di.data.lep;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import net.schmizz.sshj.SSHClient;
+import net.schmizz.sshj.transport.verification.PromiscuousVerifier;
+import net.schmizz.sshj.xfer.FileSystemFile;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import software.amazon.lambda.powertools.logging.Logging;
+import software.amazon.lambda.powertools.tracing.Tracing;
+import uk.gov.di.data.lep.exceptions.GroSftpException;
+import uk.gov.di.data.lep.library.config.Config;
+import uk.gov.di.data.lep.library.dto.GroFileLocations;
+import uk.gov.di.data.lep.library.services.AwsService;
+
+import java.io.IOException;
+import java.nio.file.Files;
+
+public class PullFile implements RequestHandler<Object, GroFileLocations> {
+    protected static Logger logger = LogManager.getLogger();
+    private final AwsService awsService;
+    private final Config config;
+    private final String groFileName;
+
+    public PullFile() {
+        this(new AwsService(), new Config());
+    }
+
+    public PullFile(AwsService awsService, Config config) {
+        this.awsService = awsService;
+        this.config = config;
+
+        groFileName = "dept_d_date.xml";
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public GroFileLocations handleRequest(Object event, Context context) {
+        logger.info("Pulling file {} from GRO", groFileName);
```

Comment:
I think we may want to be more intelligent here, e.g. "we've retrieved these files already but this one is new" - I'm happy enough if that's for the "what's the actual implementation" rather than now though